pkg/sendconfig/sendconfig.go

```diff
@@ -186,3 +188,33 @@ func onUpdateDBMode(ctx context.Context,
 	}
 	return nil
 }
+
+// -----------------------------------------------------------------------------
+// Sendconfig - Logging & Reporting Helper Functions
+// -----------------------------------------------------------------------------
+
+var (
+	latestReportedSHA []byte
+	shaLock           sync.RWMutex
+)
+
+// hasUpdateAlreadyBeenReported is a helper function to allow
+// sendconfig internals to be aware of the last logged/reported
+// update to the Kong Admin API. Given the most recent update SHA,
+// it will return true/false whether or not that SHA has previously
+// been reported (logged, e.t.c.) so that the caller can make
+// decisions (such as staggering or stifling duplicate log lines).
+//
+// TODO: This is a bit of a hack for now to keep backwards compat,
+//       but in the future we might configure rolling this into
+//       some object/interface which has this functionality as an
+//       inherent behavior.
+func hasUpdateAlreadyBeenReported(latestUpdateSHA []byte) bool {
```

Comment:
suggest a more meaningful function name SHAUpdated()
please add a unit test for the new function, so that codeCov will not complain.

Reply:
The unit tests were added :+1: 

Can you help me to better understand what you consider to be more meaningful?

To me it feels like `SHAUpdated` conveys a lot less about what the function does than `hasUpdateAlreadyBeenReported`. How would you feel about something like this instead?

```suggestion
func hasSHAUpdateAlreadyBeenReported(latestUpdateSHA []byte) bool {
```

Reply:
I think "hasSHAUpdateAlreadyBeenReported" is meaningful enough.